lib/puppet_x/chocolatey/chocolatey_install.rb

```diff
@@ -31,33 +35,32 @@ def self.install_path
       end
 
       def self.temp_dir
-        if Puppet::Util::Platform.windows?
-          require 'win32/registry'
-  
+        return unless Puppet::Util::Platform.windows?
+        require 'win32/registry'
+
+        value = nil
+        begin
+          # looking at current user may likely fail because it's likely going to be LocalSystem
+          hive = Win32::Registry::HKEY_CURRENT_USER
+          hive.open('Environment', Win32::Registry::KEY_READ | 0x100) do |reg|
+            value = reg['TEMP']
+          end
+        rescue Win32::Registry::Error
```

Comment:
This rescue should probably be logged when the registry key cannot be found or read.